tests/integration/components/file/report-btn-test.js

```diff
@@ -0,0 +1,227 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, click } from '@ember/test-helpers';
+import { setupIntl } from 'ember-intl/test-support';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { hbs } from 'ember-cli-htmlbars';
+import Service from '@ember/service';
+import { underscore } from '@ember/string';
+import { REPORT } from 'irene/utils/constants';
+import dayjs from 'dayjs';
+import { Response } from 'miragejs';
+
+class RealtimeStub extends Service {
+  ReportCounter = 0;
+}
+
+class NotificationsStub extends Service {
+  errorMsg = null;
+  successMsg = null;
+  error(msg) {
+    return (this.errorMsg = msg);
+  }
+  success(msg) {
+    return (this.successMsg = msg);
+  }
+}
+
+function serialize(payload) {
+  const serializedPayload = {};
+  Object.keys(payload.attrs).map((_key) => {
+    serializedPayload[underscore(_key)] = payload[_key];
+  });
+  return serializedPayload;
+}
+
+function serializeAll(data) {
+  return {
+    count: data.length,
+    next: null,
+    previous: null,
+    results: data.models.map((d) => {
+      return serialize(d);
+    }),
+  };
+}
+
+module('Integration | Component | file/report-btn', function (hooks) {
+  setupRenderingTest(hooks);
+  setupMirage(hooks);
+  setupIntl(hooks);
+
+  hooks.beforeEach(async function () {
+    this.owner.register('service:realtime', RealtimeStub);
+    this.owner.register('service:notifications', NotificationsStub);
+  });
+
+  test('it should render btn in file details', async function (assert) {
+    this.server.get('v2/files/:id/reports', () => {
+      return [];
+    });
+    this.set('file', {});
+    await render(hbs`<File::ReportBtn @file={{this.file}}/>`);
+    assert.dom(`[data-test-report="file-details"]`).exists();
+    assert.dom(`[data-test-report="analysis"]`).doesNotExist();
+  });
+
+  test('it should handle generating status', async function (assert) {
+    var fileReport = this.server.create('file-report', { progress: -1 });
+    this.server.get('v2/files/:id/reports', () => {
+      return [serialize(fileReport)];
+    });
+
+    this.server.post('v2/files/:id/reports', () => {
+      fileReport.update('progress', 30);
+      return serialize(fileReport);
+    });
+    this.realtimeService = this.owner.lookup('service:realtime');
+    this.set('file', { id: 1, canGenerateReport: true, isStaticDone: true });
+    await render(hbs`<File::ReportBtn @file={{this.file}}/>`);
+
+    assert
+      .dom(`[data-test-report="action-btn-label"]`)
+      .hasText(`t:generateReport:()`);
+
+    await click(`[data-test-report="action-btn-label"]`);
+    this.realtimeService.incrementProperty('ReportCounter');
+    assert
+      .dom(`[data-test-report="action-btn-label"]`)
+      .hasText(`t:generatingReport:()`);
+    assert.dom(`i.fa-spinner`).exists();
```

Comment:
Use data test selector for this instead of direct tag/class access. (That will help even if we change the icon pack in future.)